### PR TITLE
Add complex support to get buffer

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Code/Common/include/sitkImage.h [hunk 1]
```diff
@@ -400,8 +400,11 @@ namespace simple
      *
      * The pointer to the buffer is not referenced
      * counted. Additionally, while this image is made unique before
-     * returnign the pointer, additional copying and usage may
-     * introduce unexpected aliasing.
+     * returning the pointer, additional copying and usage may
+     * introduce unexpected aliasing if the image's buffer.
```
Comment (zivy) [inline Code/Common/include/sitkImage.h]: Sentence seems to end unexpectedly, probably should end "additional copying and usage may introduce unexpected aliasing."

File: Testing/Unit/Python/sitkNumpyArrayConversionTest.py [hunk 2]
```diff
@@ -175,6 +175,43 @@ def test_isVector(self):
         self.assertEqual(img.GetPixel([0,0,1,0]), (126,127))
 
 
+    def test_complex_image_to_numpy(self):
+        """Test converting back and forth between numpy and SimpleITK
+        images were the SimpleITK image is of complex type."""
```
Comment (zivy) [inline Testing/Unit/Python/sitkNumpyArrayConversionTest.py]: were -> where

File: Testing/Unit/sitkImageTests.cxx [hunk 3]
```diff
@@ -1463,6 +1459,22 @@ TEST_F(Image, GetBuffer)
...
   ASSERT_ANY_THROW( img.GetBufferAsFloat() ) << " Get with wrong type";
 
+  img = sitk::Image( 10, 10, sitk::sitkComplexFloat32 );
+  EXPECT_EQ( img.GetBufferAsFloat()[199], 0.0f ) << " Get last element in buffer ";
+  ASSERT_ANY_THROW( img.GetBufferAsInt16() ) << " Get with wrong type";
+  ASSERT_ANY_THROW( img.GetBufferAsUInt16() ) << " Get with wrong type";
+  ASSERT_ANY_THROW( img.GetBufferAsInt32() ) << " Get with wrong type";
+  ASSERT_ANY_THROW( img.GetBufferAsDouble() ) << " Get with wrong type";
+
+  img = sitk::Image( 10, 10, sitk::sitkFloat64 );
```
Comment (zivy) [inline Testing/Unit/sitkImageTests.cxx]: sitk::sitkFloat64 -> sitk::sitkComplexFloat64

File: Code/Common/include/sitkImage.h (hunk 1 above)
Comment (blowekamp) [reply Code/Common/include/sitkImage.h]: if->of

Comment (dave3d) [inline Code/Common/include/sitkImage.h]: if the image's buffer ... what?

Comment (blowekamp) [reply Code/Common/include/sitkImage.h]: if -> of

**Comment by dave3d** [comment]
So I downloaded and built this patch, and it compiles and the tests pass.

But I tried a simple python code

```
import SimpleITK as sitk
img = sitk.Image(100,100,sitk.sitkComplexFloat32)
print(img)
x = sitk.GetArrayFromImage(img)

```
and I get the old error message:
`RuntimeError: Images of Complex Pixel types currently are not supported.
`
I checked sitkNumpyArrayConversion.cxx, and it's the new code, so that message is gone.  So I don't know what's happening.  It's probably me.

**Comment by blowekamp** [comment]
> So I downloaded and built this patch, and it compiles and the tests pass.
> 
> But I tried a simple python code
> 
> ```
> import SimpleITK as sitk
> img = sitk.Image(100,100,sitk.sitkComplexFloat32)
> print(img)
> x = sitk.GetArrayFromImage(img)
> ```
> 
> and I get the old error message:
> `RuntimeError: Images of Complex Pixel types currently are not supported. `
> I checked sitkNumpyArrayConversion.cxx, and it's the new code, so that message is gone. So I don't know what's happening. It's probably me.

You likely are not loading the versions of SimpleITK you just compiled try `print(sitk.__file__)` and `print(sitk.Version())`

**Comment by dave3d** [comment]
```
>>> print(sitk.__file__)
/Users/dchen/test/SimpleITK-build/Testing/Installation/PythonVirtualenv/lib/python3.7/site-packages/SimpleITK-1.3.0.dev216+ga642b-py3.7-macosx-10.7-x86_64.egg/SimpleITK/__init__.py
>>> print(sitk.Version())
SimpleITK Version: 1.3.0.dev216-ga642b (ITK 4.13)
Compiled: Jun 13 2019 12:51:39

>>> x = sitk.GetArrayFromImage(img)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dchen/test/SimpleITK-build/Testing/Installation/PythonVirtualenv/lib/python3.7/site-packages/SimpleITK-1.3.0.dev216+ga642b-py3.7-macosx-10.7-x86_64.egg/SimpleITK/SimpleITK.py", line 3412, in GetArrayFromImage
    arrayView = GetArrayViewFromImage(image)
  File "/Users/dchen/test/SimpleITK-build/Testing/Installation/PythonVirtualenv/lib/python3.7/site-packages/SimpleITK-1.3.0.dev216+ga642b-py3.7-macosx-10.7-x86_64.egg/SimpleITK/SimpleITK.py", line 3399, in GetArrayViewFromImage
    imageMemoryView =  _SimpleITK._GetMemoryViewFromImage(image)
RuntimeError: Images of Complex Pixel types currently are not supported.

```

**Comment by blowekamp** [comment]
> >>> print(sitk.__file__)
> /Users/dchen/test/SimpleITK-build/Testing/Installation/PythonVirtualenv/lib/python3.7/site-packages/SimpleITK-1.3.0.dev216+ga642b-py3.7-macosx-10.7-x86_64.egg/SimpleITK/__init__.py
> >>> print(sitk.Version())
> SimpleITK Version: 1.3.0.dev216-ga642b (ITK 4.13)
> Compiled: Jun 13 2019 12:51:39
> 

The end of the version should have the hash "db6a0810aa3af8cae7c6c39e7674834b1ffdd162".

**Comment by dave3d** [comment]
LGTM

**Comment by blowekamp** [comment]
@richardbeare How does R deal with complex numbers? Can that get buffer method be updates to support these types too?

**Pull request merged**